usr/sbin/rear

```diff
@@ -777,6 +777,16 @@ Debug "Sourced files must be below one of the default TRUSTED_PATHS: ${TRUSTED_P
 for script in $SHARE_DIR/lib/[a-z]*.sh ; do
     Source $script || BugError "Failed to source $script"
 done
+# In simulation mode the workflow scripts $SHARE_DIR/lib/[a-z]*-workflow.sh need to be executed (sourced)
+# to get the WORKFLOW_... functions defined, otherwise the "Check for and run the requested workflow" code below
+# could not run the requested workflow, see https://github.com/rear/rear/pull/3424#issuecomment-2747079736
+# (in simulation mode a WORKFLOW_... function is run but its called scripts are not executed by the Source function):
+if test "$SIMULATE" ; then
+    Debug "In simulation mode the WORKFLOW_... functions are needed so the 'lib/[a-z]*-workflow.sh' scripts get executed"
+    for script in $SHARE_DIR/lib/[a-z]*-workflow.sh ; do
```

Comment:
Idea: move the workflow files into `$SHARE_DIR/lib/workflows/<name>.sh` and always `builtin source` it.

That way you can get rid of the special handling around simulation mode here.

Reply:
After talking about again, actually it might be even better to "fix" the simulation mode check to *not* simulate files under `$SHARE_DIR/lib` and that allows getting rid of this entire special handling here.

Reply:
And then we realize that `$SHARE_DIR/conf` files are currently simulated but they shouldn't, same as the `$CONF_DIR` files are not simulated.

Therefore here is another idea:

introduce a new function to decide simulation mode per script file and have there more complex checks for what to simulate and not:

```
    if is_simulation_mode_for "$source_file" ; then
    #if test "$SIMULATE" && expr "$source_file" : "$SHARE_DIR" >/dev/null; then
        LogPrint "Source $relname"
        return
    fi
```

And in that `is_simulation_mode_for` function we could even use a `case` to quickly match paths to find out if a script should be simulated or not.

A nice optimization would be to not use `expr` to determine simulation mode as that is an external program.

Reply:
To be done as time permits via
https://github.com/rear/rear/issues/3438